driver-core/src/test/unit/com/mongodb/internal/connection/ScramShaAuthenticatorSpecification.groovy

```diff
@@ -387,6 +387,24 @@ class ScramShaAuthenticatorSpecification extends Specification {
         async << [true, false]
     }
 
+    def 'should skip empty exchange on server authentication SHA-256'() {
+        given:
+        def serverResponses = createMessages('''
+            S: r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,s=W22ZaJ0SNY7soEsUEjb6gQ==,i=4096
+            S: v=6rriTRBi23WpRR/wtup+mMhUZUn/dB5nLTJRsjl95G4=
+        ''').last()
+        def authenticator = new ScramShaAuthenticator(SHA256_CREDENTIAL, { 'rOprNGfwEbeRWgbNEkqO' }, { 'pencil' })
+
+        when:
+        authenticate(createConnection(serverResponses, 1), authenticator, async)
+
+        then:
+        noExceptionThrown()
+
+        where:
+        async << [true, false]
```

Comment:
Can we add to this where clause to also test SHA1 credential?  You can use `.combinations` to test all the scenarios

Reply:
I have updated each SHA1 and SHA256 test that should complete successfully to call a method that will run with and without an empty exchange.